src/jsonschema/package_converter.py

```diff
@@ -0,0 +1,126 @@
+        elif package_property == PackageProperty.COMMENT:
+            return package.comment
+        elif package_property == PackageProperty.COPYRIGHT_TEXT:
+            return apply_if_present(str, package.copyright_text)
+        elif package_property == PackageProperty.DESCRIPTION:
+            return package.description
+        elif package_property == PackageProperty.DOWNLOAD_LOCATION:
+            return str(package.download_location)
+        elif package_property == PackageProperty.EXTERNAL_REFS:
+            return [self.external_package_ref_converter.convert(external_ref) for external_ref in
+                    package.external_references] or None
+        elif package_property == PackageProperty.FILES_ANALYZED:
+            return package.files_analyzed
+        elif package_property == PackageProperty.HAS_FILES:
+            package_contains_file_ids = [relationship.related_spdx_element_id for relationship in
+                                         find_package_contains_file_relationships(document, package)]
+            file_contained_in_package_ids = [relationship.spdx_element_id for relationship in
+                                             find_file_contained_by_package_relationships(document, package)]
+            return package_contains_file_ids + file_contained_in_package_ids or None
+        elif package_property == PackageProperty.HOMEPAGE:
+            return apply_if_present(str, package.homepage)
+        elif package_property == PackageProperty.LICENSE_COMMENTS:
+            return package.license_comment
+        elif package_property == PackageProperty.LICENSE_CONCLUDED:
+            return apply_if_present(str, package.license_concluded)
+        elif package_property == PackageProperty.LICENSE_DECLARED:
+            return apply_if_present(str, package.license_declared)
+        elif package_property == PackageProperty.LICENSE_INFO_FROM_FILES:
+            if isinstance(package.license_info_from_files, list):
+                return [str(license_expression) for license_expression in package.license_info_from_files] or None
+            return apply_if_present(str, package.license_info_from_files)
+        elif package_property == PackageProperty.NAME:
+            return package.name
+        elif package_property == PackageProperty.ORIGINATOR:
+            if isinstance(package.originator, Actor):
+                return package.originator.to_serialized_string()
+            return apply_if_present(str, package.originator)
+        elif package_property == PackageProperty.PACKAGE_FILE_NAME:
+            return package.file_name
+        elif package_property == PackageProperty.PACKAGE_VERIFICATION_CODE:
+            return apply_if_present(self.package_verification_code_converter.convert, package.verification_code)
+        elif package_property == PackageProperty.PRIMARY_PACKAGE_PURPOSE:
+            return package.primary_package_purpose.name if package.primary_package_purpose is not None else None
```

Comment:
`OPERATING_SYSTEM` must be converted to OPERATING-SYSTEM

Reply:
The same goes for `ExternalPackageRefProperty.REFERENCE_CATEGORY`, where `PACKAGE_MANAGER` and `PERSISTENT_ID` have to be converted

Reply:
sigh... I believe I came across https://github.com/spdx/spdx-spec/blob/development/v2.3.1/schemas/spdx-schema.json#L325, then read https://github.com/spdx/spdx-spec/issues/792, and concluded that by now, both should be supported. Apparently, this is not true for all enums...

Reply:
Wait, I am seeing ghosts now... For `PACKAGE_MANAGER` and `PERSISTENT_ID`, both are supported now according to the spec: https://github.com/spdx/spdx-spec/blob/development/v2.3.1/schemas/spdx-schema.json#L325
For `OPERATING_SYSTEM`, only underscore is supported: https://github.com/spdx/spdx-spec/blob/development/v2.3.1/schemas/spdx-schema.json#L411

Do you see the same things? I could have sworn I saw something different when I looked at it before :sweat_smile: . Anyway, I will remove the commit I made again

Reply:
Yeah, seems like both are to be supported now, so this should be fine